src/builtins/BuiltinTemporal.cpp

```diff
@@ -60,11 +60,48 @@ static Value builtinTemporalPlainDateConstructor(ExecutionState& state, Value th
     if (!newTarget.hasValue()) {
         ErrorObject::throwBuiltinError(state, ErrorObject::TypeError, ErrorObject::Messages::New_Target_Is_Undefined);
     }
-    Value y = Value(argv[0].asInt32());
-    Value m = Value(argv[1].asInt32());
-    Value d = Value(argv[2].asInt32());
     Value calendar = TemporalCalendar::toTemporalCalendarWithISODefault(state, argc > 4 ? argv[3] : Value());
-    return TemporalPlainDate::createTemporalDate(state, y, m, d, calendar, newTarget);
+    return TemporalPlainDate::createTemporalDate(state, argv[0].asInt32(), argv[1].asInt32(), argv[2].asInt32(), calendar, newTarget);
+}
+
+static Value builtinTemporalPlainDateFrom(ExecutionState& state, Value thisValue, size_t argc, Value* argv, Optional<Object*> newTarget)
+{
+    if (argc > 1 && !argv[1].isObject()) {
+        ErrorObject::throwBuiltinError(state, ErrorObject::TypeError, "options must be object");
+    }
+    Object* optionsObject = argc > 1 ? argv[1].asObject() : new Object(state);
```

Comment:
Is there a simple way to avoid creation of objects, or it would complicate the code a lot?